src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+AuthorizationOTPElsTests.swift

```diff
@@ -406,4 +406,50 @@ final class HTTPClientAuthorizationOTPElsTests: CWATestCase {
 		XCTAssertNotNil(expectedOtpError)
 		XCTAssertEqual(expectedOtpError, .invalidResponseError)
 	}
+	
+	func testGIVEN_AuthorizeOTP_WHEN_NoNetworkConnection_THEN_NetworkErrorReturned() {
+		// GIVEN
+		let url = URL(staticString: "https://localhost:8080")
+		let notConnectedError = NSError(
+			domain: NSURLErrorDomain,
+			code: NSURLErrorNotConnectedToInternet,
+			userInfo: nil
+		)
+
+		let session = MockUrlSession(
+			data: nil,
+			nextResponse: HTTPURLResponse(
+				url: url,
+				statusCode: 500,
+				httpVersion: nil,
+				headerFields: nil
+			),
+			error: notConnectedError
+		)
+
+		let stack = MockNetworkStack(
+			mockSession: session
+		)
+
+		let expectation = self.expectation(description: "completion handler is called without an error")
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		var expectedOtpError: OTPError?
+		HTTPClient.makeWith(mock: stack).authorize(otpEls: otp, ppacToken: ppacToken, completion: { result in
+			switch result {
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				expectedOtpError = otpError
+			}
+			expectation.fulfill()
+		})
+
+		// THEN
+		waitForExpectations(timeout: expectationsTimeout)
+		XCTAssertNotNil(expectedOtpError)
+		XCTAssertEqual(expectedOtpError, .noNetworkConnection)
```

Comment:
Why no check in place?
```suggestion
```

Reply:
I originally implemented this and will explain why i choosed the original style: I find it easier to read. When putting the assert check in the completion handler, one can easily oversee this. When it is at the end, one will find it faster because everyone expects it at the end. Also the Given-When-Then pattern is broken because the assert is "inside" the Then-check.
You can let the code now as it is or revert it, your choice.